### PR TITLE
has-transformed-position

### PR DESCRIPTION
**Pull request by ddkohler**

### CODE REVIEW

File: yeps/yep-312.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+---
+yep: 312
+title: has-reference-position
```
Comment (ksunden) [inline yeps/yep-312.md:3]: update name
```suggestion
title: has-transformed-position
```

File: yeps/yep-312.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+author: Dan Kohler <ddkohler@wisc.edu>
+status: draft
```
Comment (ksunden) [inline yeps/yep-312.md:5]:
```suggestion
status: accepted
```

I'm leaning towards just accepting this off the bat, any qualms @untzag?

File: yeps/yep-312.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+* control_kind : "hinted"
+* record_kind : "metadata"
+
+### native_position
+
+* getter : "get_native_position"
+* setter : "set_native_position"
+* limits_getter : "get_native_limits"
+* units_getter : "get_native_units"
+* control_kind : "hinted"
```
Comment (ksunden) [inline yeps/yep-312.md:143]:
```suggestion
* control_kind : "normal"
```

see above

File: yeps/yep-312.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+# Backwards Compatibility
+
+This trait has no effect on `has-limits` when `reference_position = 0.0` and `to_transformed(x) == to_native(x) == x`.
+
+In state, all values relating to position (`position`, `destination`, `native_reference_position` and `hw_limits`) are stored in native coordinates.
+This is so that state values will remain valid even when transforms change.
+
+# Rejected Ideas
+
```
Comment (ksunden) [inline yeps/yep-312.md:161]: some mention of the ideas we went through to get here may be useful context here

**Comment by ddkohler** [comment]
I should add a limits getter for native position.  Please wait to merge

**Comment by ddkohler** [comment]
Current leaning is the "omit" type.  But I have a bias to over-emphasize smaller filesizes.  

I am much more comfortable with this option if can I assume daemon-specific transform parameters are properties and use "metadata" record_type, so there is hope to invert the transformed position to the native for those (rare) cases where it matters.  Perhaps we can put that in the yep?

**Review by untzag — APPROVED** [review]
looks good enough to merge, thanks for the write up @ddkohler

**Comment by ksunden** [comment]
There was an open question about "omitted" vs "metadata"

I am personally fine with either, I think metadata makes sense and I think it should be assumed that those parameters are not changing mid acquisition.

**Pull request merged**